h2olog.cc

```diff
@@ -85,6 +85,40 @@ static void show_process(pid_t pid)
     fprintf(stderr, "Attaching pid=%d (%s)\n", pid, cmdline);
 }
 
+static std::string join_str(const std::string &sep, const std::vector<std::string> &strs)
+{
+    std::string s;
+    for (auto iter = strs.cbegin(); iter != strs.cend(); ++iter) {
+        if (iter != strs.cbegin()) {
+            s += sep;
+        }
+        s += *iter;
+    }
+    return s;
+}
+
+static std::string generate_header_filter_cflag(const std::vector<std::string> &tokens)
+{
+    std::vector<std::string> conditions;
+
+    for (auto &token : tokens) {
+        char buf[256];
+        snprintf(buf, sizeof(buf), "/* %s */ (slen) == %lu", token.c_str(), token.size());
```

Comment:
You should use `z`(`size_t`, `zd=ssize_t`, `zu=size_t`) modifier instead of `l`(`long`) for `size_t`

https://github.com/toru/h2olog/blob/2d7e803a41f36dba7573a0887d9bdd3a195d1e45/h2olog.cc#L110

Ditto

Reference
- https://linux.die.net/man/3/printf